src/main/java/com/gmail/yuramitryahin/config/AppConfig.java

```diff
@@ -4,22 +4,29 @@
 import java.util.Properties;
 import javax.sql.DataSource;
 import org.apache.commons.dbcp2.BasicDataSource;
+import org.springframework.beans.factory.annotation.Autowired;
 import org.springframework.context.annotation.Bean;
 import org.springframework.context.annotation.ComponentScan;
 import org.springframework.context.annotation.Configuration;
+import org.springframework.context.annotation.PropertySource;
+import org.springframework.core.env.Environment;
 import org.springframework.orm.hibernate5.LocalSessionFactoryBean;
 
 @Configuration
+@PropertySource("classpath:db.properties")
 @ComponentScan(basePackages = {"com.gmail.yuramitryahin.dao",
         "com.gmail.yuramitryahin.service"})
 public class AppConfig {
+    @Autowired
+    private Environment env;
```

Comment:
What is better: setter, constructor, or field injection? Think why.

Reply:
constructor